.github/workflows/emscripten.yml

```diff
@@ -9,11 +9,10 @@ on:
     branches:
       - main
       - maintenance/**
```

Comment:
Is there a reason to keep the pull request trigger? Since we have `if: github.repository == 'scikit-image/scikit-image'` enabled below, a pull request targeting either `main` or `maintenance` would still be skipped.

Reply:
Ideally, we'll want that the out-of-tree builds will run on every PR similar to standard PR checks, in order to catch errors early/during PRs. I'm not sure I follow, though. The condition `if: github.repository == 'scikit-image/scikit-image'` will stop the workflow from running on pull requests on forks, but it will allow running on PRs to `scikit-image` regardless of where the base branch of the PR is (whether it's coming from a fork, or a collaborator's branch for those who have write access to the repo). A build triggered by a pull request to `scikit-image` should not get skipped when targeting either `main` or `maintenance`, as far as I understand.

Reply:
Ah, I think I was confused. Then, is there any reason not to give it the same permissions as the `test.yml` workflow using 
```yml
on: [push, pull_request, merge_group]
```

And there's also no reason to keep the `if: github.repository == 'scikit-image/scikit-image'` right?

Reply:
Yes, there's no reason as such, it's mostly because I have been implementing the workflow in the same way across projects, and I guess skipping the workflow on forks saves time with an instant notification :) Let me remove this and synchronise the `test.yml` workflow and this one in the next commit.

Reply:
Addressed in 81d1eabea53d3e22ea1b71ed5987e431fa6a0348.

Reply:
I don't follow, in what way does this save time? :thinking:

Reply:
and in 67aa8ee36d2f3c40a82c5f82e9539741063122a0.

Reply:
> I don't follow, in what way does this save time? 🤔

Ah, I worded that a bit wrong. If one syncs their `main` branch with the upstream `main` branch, and that triggers the workflow runs, it would skip the workflow instantly. Skipping workflows would also be useful for developers who are trying to test out something on their forks so that they don't have to disable the workflows manually. Though, I would agree that this is a niche use case and my reasoning is built upon other projects that skip workflows.